tests/collections/conftest.py

```diff
@@ -16,26 +16,6 @@
 
 import pennylane as qml
 
-try:
-    import torch
-except ImportError as e:
-    torch = None
-
-try:
-    import tensorflow as tf
-
-    if tf.__version__[0] == "1":
-        print(tf.__version__)
-        import tensorflow.contrib.eager as tfe
-
-        tf.enable_eager_execution()
-        Variable = tfe.Variable
-    else:
-        from tensorflow import Variable
-except ImportError as e:
-    tf = None
-    Variable = None
-
```

Comment:
The motivation for doing this is:
- A new `conftest` was introduced in the `tests/qnn` folder
- This resulted in failing tests
- The tests were failing because `tests/collections/test_collections.py` and `tests/collections/test_qnode_collection` include a `from conftest import torch, tf, Variable` line.
- Importing from conftest is not recommended because it can be ambiguous which one to import from (https://docs.pytest.org/en/2.7.3/plugins.html?highlight=re):
![image](https://user-images.githubusercontent.com/49409390/80727042-3a5b0e80-8ad3-11ea-963a-0f2ea459c3a7.png)
NOTE: This is from an older version of pytest, a more recent reference couldn't be found
- It made sense to make this change rather than simply removing the `tests/qnn/conftest.py` because importing conftest is not recommended.
- The result is a few more lines of code in `tests/collections/test_collections.py` and `tests/collections/test_qnode_collection`. Maybe we could do something smarter but it seems ok for now.

Reply:
@trbromley, the `__init__.py` didn't fix it? 🤔 I also wonder if `from .conftest import ...` would fix it.

Reply:
Ah just revisited this and tried again:
- If we add `__init__.py` to `tests/collections` AND `tests/`, as well as update to `from .conftest`, the `test_collections`/`test_qnode_collection` will work
- However, the full test suite stops working: I get an error for the tests running `from gate_data import ...`
- This can be fixed by adding an `__init__.py` to all the required subfolders and changing to `from .gate_data import ...`
- I went a little bit down that route but there's so many files to change, it made me think that the solution currently adopted with code duplication is the lesser of two evils.